portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -75,6 +78,13 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         // Get the input from the form.
         String name = getParameter(request, "name", "");
         String text = getParameter(request, "comment", "");
+
+        Entity commentEntity = new Entity("Comment");
+        commentEntity.setProperty("name", name);
+        commentEntity.setProperty("text", text);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(commentEntity);
         
         Comment comment = new Comment(name, text);
         comments.add(comment);
```

Comment:
You should remove the `comments` List now that you have the datastore -- always load the latest comments from there